libkbfs/folder_branch_ops.go

```diff
@@ -5950,6 +5950,7 @@ func (fbo *folderBranchOps) unstageAfterFailedResolution(ctx context.Context,
 	default:
 	}
 
+	ctx = newLinkedContext(ctx)
```

Comment:
I'll add an explanation to the code, but:
1) If the error is caused by a cancelled context then the recovery (archiving) will need to use a separate context anyways.
2) In such cases we have to be very careful where the error occurs and what to archive, making that solution more complicated.
3) The other "common" error case is losing server connection and after detecting that we won't have much luck archiving things.

Thus this seems like the simplest way to avoid missing the blocks.